app/setup/post.controller.tests.js

```diff
@@ -53,21 +61,233 @@ describe('setup post controller', () => {
           'address-line2': formValues.addressLine2,
           'city': formValues.city,
           'postcode': formValues.postcode,
-          'email': formValues.email })
+          'email': formValues.email
+        })
         .set('Cookie', cookieHeader)
         .set('Content-Type', 'application/json')
         .set('Accept', 'application/json')
         .end((err, res) => {
-          response = res
+          responseRedirect = res
           done(err)
         })
     })
+
     it('should redirect to /confirm', () => {
-      expect(response.statusCode).to.equal(303)
+      expect(responseRedirect.statusCode).to.equal(303)
     })
+
     it('should redirect to the insert direct debit details page', () => {
-      let url = confirmation.paths.index.replace(':paymentRequestExternalId', paymentRequestExternalId)
-      expect(response.header).property('location').to.equal(url)
+      const url = confirmation.paths.index.replace(':paymentRequestExternalId', paymentRequestExternalId)
+      expect(responseRedirect.header).property('location').to.equal(url)
+    })
+  })
+
+  describe('should keep the field values when submitting the form with validation errors', () => {
+    const paymentRequestExternalId = 'wsfihsdufh2f'
+    const paymentRequest = paymentFixtures.validPaymentRequest({
+      external_id: paymentRequestExternalId
+    })
+    let cookieHeader
+    let $
+    const formValues = {
+      accountHolderName: 'Mr T',
+      sortCode: '12 34 567',
+      accountNumber: '1234567',
+      requiresAuthorisation: 'false',
+      city: 'Sofia',
+      countryCode: 'ES',
+      postcode: 'W1 3EF',
+      addressLine1: 'address line 1',
+      addressLine2: 'address line 2',
+      email: 'payer@willpay.co.uk'
```

Comment:
Please don’t use email address domains that might actually be registered! Use something ending in `.test` (guaranteed to never resolve)

Reply:
Thanks @alexbishop1 ! Addressed in [5020dea]